batchcloser/closers.go

```diff
@@ -0,0 +1,68 @@
+package batchcloser
+
+import (
+	"context"
+	"io"
+
+	"github.com/reddit/baseplate.go/batcherror"
+)
+
+// CancelCloser can be used to wrap context.CancelFunc in a struct that
+// implements io.Closer.
+type CancelCloser struct {
+	CancelFunc context.CancelFunc
+}
+
+// Close implements io.Closer by calling the given CancelFunc and returning nil.
+func (c CancelCloser) Close() error {
+	c.CancelFunc()
+	return nil
+}
+
+// SimpleCloser can be used to wrap a function with the same signature as
+// the "Close" function defined in io.Closer in a struct that implements io.Closer.
+type SimpleCloser struct {
+	CloseFunc func() error
+}
+
+// Close implements io.Closer by calling the given CloseFunc.
+func (c SimpleCloser) Close() error {
+	return c.CloseFunc()
+}
+
+// NewBatchCloser returns a new *BatchCloser initialized with the given io.Closer
+// objects.
+func NewBatchCloser(closers ...io.Closer) *BatchCloser {
```

Comment:
Also it's probably worth mention that just declare `var closers batchcloser.BatchCloser` is good enough and this is optional. Since `BatchCloser.Add` already support vararg maybe we don't really need this one? it's basically sugar to combine 2 lines into 1.

Reply:
Ya, I'm just gonna get rid of it.

Reply:
I think you have to do `closers := &batchcloser.BatchCloser{}`?  Which also isn't crazy.

Reply:
if you do `var closers batchcloser.BatchCloser`, calling `closers.Add(...)` and `closers.Close()` still work, it's just that `closer` doesn't implement `io.Closer` (for that you need to use `&closers` instead, but if you make the `Close` function's receiver from pointer to reference then that is also not an issue any more).

Reply:
(but technically `Close` is a mutation call so leave that as pointer receiver is probably better)